toolchain/sem_ir/block_value_store.h

```diff
@@ -82,6 +102,37 @@ class BlockValueStore : public Yaml::Printable<BlockValueStore<IdT>> {
   }
 
  private:
+  // A canonical block, for which we allocate a deduplicated ID.
+  struct CanonicalBlock {
+    // This is mutable so we can repoint it at the allocated data if insertion
+    // succeeds.
+    mutable llvm::ArrayRef<ElementType> data;
+
+    friend auto CarbonHashValue(CanonicalBlock block, uint64_t seed)
+        -> HashCode {
+      Hasher hasher(seed);
+      hasher.HashSizedBytes(block.data);
+      return static_cast<HashCode>(hasher);
+    }
+  };
+
+  struct CanonicalBlockDenseMapInfo {
+    static constexpr ElementType SpecialData[1] = {ElementType::Invalid};
+    static auto getEmptyKey() -> CanonicalBlock {
+      return CanonicalBlock{{SpecialData, SpecialData}};
+    }
+    static auto getTombstoneKey() -> CanonicalBlock {
+      return CanonicalBlock{{SpecialData, SpecialData + 1}};
```

Comment:
I was trying to figure out what's happening here, would be below be okay/equivalent?

```suggestion
      return CanonicalBlock{llvm::ArrayRef(SpecialData, 0)};
    }
    static auto getTombstoneKey() -> CanonicalBlock {
      return CanonicalBlock{llvm::ArrayRef(SpecialData, 1)};
```

Reply:
Done.